opentelemetry-instrumentation/src/opentelemetry/instrumentation/_blobupload/utils/simple_blob_uploader_adaptor.py

```diff
@@ -0,0 +1,138 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Provides 'blob_uploader_from_simple_blob_uploader', a utility for
+backend providers to more easily provide a full-fledged BlobUploader
+by implementing the simpler 'SimpleBlobUploader' interface.
+
+The 'blob_uploader_from_simple_blob_uploader' utility takes care of
+common machinery such as scheduling, retries, background uploading,
+etc. allowing providers of specific BlobUploader backends to supply
+a simpler set of synchronous uploading instructions.
+"""
+
+import atexit
+import logging
+from concurrent.futures import Executor, ThreadPoolExecutor
+from typing import Optional
+
+from opentelemetry.instrumentation._blobupload.api import (
+    Blob,
+    BlobUploader,
+    detect_content_type,
+)
+from opentelemetry.instrumentation._blobupload.utils.simple_blob_uploader import (
+    SimpleBlobUploader,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+def _with_content_type(blob: Blob) -> Blob:
+    """Returns a variant of the Blob with the content type auto-detected if needed."""
+    if blob.content_type is not None:
+        return blob
+    content_type = detect_content_type(blob.raw_bytes)
```

Comment:
can't we infer the content type from the labels, instead of inspecting the bytes?